File: python/Mahasiswa.py
from SivitasAkademik import SivitasAkademik
from ProgramStudi import ProgramStudi
from Course import Course

class Mahasiswa (SivitasAkademik):
  __nim = ''# atribut NIM
  __prodi = ProgramStudi()# atribut prodi
  __fakultas = ''# atribut fakultas
  __course = []# atribut fakultas


  def __init__(self, nik = '', nama = '', gender = '', nim = '', fakultas = '', prodi = ProgramStudi(), asalUniv = '', emailEdu = '', course = Course()):
    # inisiasi dengan value awal
    super().__init__(nik, nama, gender, asalUniv, emailEdu)
    self.__nim = nim
    self.__prodi = prodi
    self.__fakultas = fakultas
    self.__course.append(course)
  

  def getNim(self):
    # return NIM mahasiswa
    return self.__nim
  
  def getProdi(self):
    # return prodi
    return self.__prodi
  
  def getFakultas(self):
    #return fakultas
    return self.__fakultas
  
  def getCourse(self):
    #return course
    return self.__course
  
  def setNim(self, nim):
    # menggati value NIM
    self.__nim = nim
  
  def setProdi(self,prodi):
    # mengganti value prodi
    self.__prodi = prodi
  
  def setFakultas(self, fakultas):
    # mengganti value fakultas
    self.__fakultas = fakultas
    
  def addCourse(self, course):
    self.__course.append(course)
  

File: c++/Mahasiswa.cpp
class Mahasiswa : public SivitasAkademik {
  private:
    string nim; // atribut NIM
    ProgramStudi prodi; // atribut prodi
    string fakultas; // atribut fakultas
    vector<Course> course; // atribut course

  public:
    Mahasiswa(string nik = "", string nama = "", string gender = "", string nim = "", string fakultas = "", ProgramStudi prodi = ProgramStudi(), string asalUniv = "", string emailEdu = "", Course course = Course()) : SivitasAkademik(nik, nama, gender, asalUniv, emailEdu) {
    this->nim = nim;
    this->prodi = prodi;
    this->fakultas = fakultas;
    this->course.push_back(course);
    }
    string getNim() {
    // return NIM mahasiswa
    return this->nim;
  }

  ProgramStudi getProdi() {
    // return prodi
    return this->prodi;
  }

  string getFakultas() {
    // return fakultas
    return this->fakultas;
  }

  vector<Course> getCourse() {
    // return course
    return this->course;
  }

  void setNim(string nim) {
    // mengganti value NIM
    this->nim = nim;
  }

  void setProdi(ProgramStudi prodi) {
    // mengganti value prodi
    this->prodi = prodi;
  }

  void setFakultas(string fakultas) {
    // mengganti value fakultas
    this->fakultas = fakultas;
  }

  void addCourse(Course course) {
    this->course.push_back(course);
  }
};